cmd/oras/root/cp.go

```diff
@@ -141,7 +141,10 @@ func runCopy(cmd *cobra.Command, opts *copyOptions) error {
 		// correct source digest
 		opts.From.RawReference = fmt.Sprintf("%s@%s", opts.From.Path, desc.Digest.String())
 	}
-	_ = opts.Printer.Println("Copied", opts.From.AnnotatedReference(), "=>", opts.To.AnnotatedReference())
+
+	if err := metadataHandler.OnCopied(opts.From.AnnotatedReference(), opts.To.AnnotatedReference(), desc); err != nil {
```

Comment:
`AnnotatedReference()` itself is a text formatted string. Therefore, this PR is not a valid handler extraction.